torch/csrc/dynamo/cache_entry.cpp

```diff
@@ -9,6 +9,7 @@ CacheEntry::CacheEntry(const py::handle& guarded_code, PyObject* backend)
   this->check_fn = guarded_code.attr("check_fn");
   this->code = guarded_code.attr("code");
   this->compile_id = guarded_code.attr("compile_id");
+  this->trace_id = PyUnicode_AsUTF8(guarded_code.attr("trace_id").ptr());
```

Comment:
You can't do this. You're not keeping trace_id attribute of guarded_code live, and guarded_code is not guaranteed to stay live beyond this constructor. Lifetime of trace_id is tied to the underlying Python object

Reply:
Since you want to format a string anyway, probably easiest to just make an owning std::string to live on the CacheEntry

Reply:
Oh I see, the other params set here will have a refcount since they are py::object but the var here can go out of scope. Addressed this in newest commit by initializing std::string as you suggested.